### PR TITLE
PaymentPage: Fix localStorage access issue in incognito

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes # (issue)
- It is known that localStorage access throws error in third party context when opened in Google Chrome Incognito Mode.
So, use our wrapper of localStorage instead of native one.

- Also, payment page is now translated server side, avoiding flash of untranslated content on the page

**Environment**: Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: packages/features/ee/payments/components/PaymentPage.tsx [hunk 1]
```diff
@@ -21,11 +22,14 @@ const PaymentPage: FC<PaymentPageProps> = (props) => {
...
   const [is24h, setIs24h] = useState(isBrowserLocale24h());
   const [date, setDate] = useState(dayjs.utc(props.booking.startTime));
+  const [timezone, setTimezone] = useState<string | null>(null);
   useTheme(props.profile.theme);
   const isEmbed = useIsEmbed();
   useEffect(() => {
     let embedIframeWidth = 0;
-    setDate(date.tz(localStorage.getItem("timeOption.preferredTimeZone") || dayjs.tz.guess()));
+    const _timezone = localStorage.getItem("timeOption.preferredTimeZone") || dayjs.tz.guess();
+    setTimezone(_timezone);
```
Comment (hariombalhara) [inline packages/features/ee/payments/components/PaymentPage.tsx:31]: Compute timezone on client side only where localStorage is available.

File: packages/features/ee/payments/components/PaymentPage.tsx [hunk 2]
```diff
@@ -95,9 +99,7 @@ const PaymentPage: FC<PaymentPageProps> = (props) => {
                         {date.format("dddd, DD MMMM YYYY")}
                         <br />
                         {date.format(is24h ? "H:mm" : "h:mma")} - {props.eventType.length} mins{" "}
-                        <span className="text-gray-500">
-                          ({localStorage.getItem("timeOption.preferredTimeZone") || dayjs.tz.guess()})
-                        </span>
+                        <span className="text-gray-500">({timezone})</span>
```
Comment (hariombalhara) [inline packages/features/ee/payments/components/PaymentPage.tsx:102]: localStorage can't be accessed server side.

File: packages/features/ee/payments/pages/payment.tsx [hunk 3]
```diff
@@ -102,6 +106,7 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
       user,
       eventType,
       booking,
+      trpcState: ssr.dehydrate(),
```
Comment (hariombalhara) [inline packages/features/ee/payments/pages/payment.tsx:109]: Does translation server side.

**Review by hariombalhara — COMMENTED** [review]
Done with self review

**Review by alishaz-polymath — APPROVED** [review]
🚀

**Pull request merged**